### PR TITLE
Improve radar minimap on isometric maps.

### PR DESCRIPTION
**Pull request by pchote**
Ingame radar counterpart to #8825.

### CODE REVIEW

**Comment by penev92** [comment]
Seems legit :+1:

**Comment by Phrohdoh** [comment]
```
Exception of type `System.Reflection.TargetInvocationException`: Exception has been thrown by the target of an invocation.
Inner Exception of type `System.ArgumentException`: Collection must not be empty.
Parameter name: ts
    at OpenRA.Exts.CompareBy[MPos,Int32] (IEnumerable`1 ts, System.Func`2 selector, Int32 modifier, Boolean throws) [0x0002a] in /Users/thill/projects/openra/OpenRA.Game//Exts.cs:208
  at OpenRA.Exts.MinBy[MPos,Int32] (IEnumerable`1 ts, System.Func`2 selector) [0x00005] in /Users/thill/projects/openra/OpenRA.Game//Exts.cs:181
  at OpenRA.Mods.Common.Widgets.RadarWidget.MapBoundsChanged () [0x000ec] in /Users/thill/projects/openra/OpenRA.Mods.Common//Widgets/RadarWidget.cs:117
  at OpenRA.Mods.Common.Widgets.RadarWidget.Initialize (OpenRA.Widgets.WidgetArgs args) [0x0004a] in /Users/thill/projects/openra/OpenRA.Mods.Common//Widgets/RadarWidget.cs:84
  at OpenRA.WidgetLoader.LoadWidget (OpenRA.Widgets.WidgetArgs args, OpenRA.Widgets.Widget parent, OpenRA.MiniYamlNode node) [0x000ef] in /Users/thill/projects/openra/OpenRA.Game//Widgets/WidgetLoader.cs:63
  at OpenRA.WidgetLoader.LoadWidget (OpenRA.Widgets.WidgetArgs args, OpenRA.Widgets.Widget parent, OpenRA.MiniYamlNode node) [0x00152] in /Users/thill/projects/openra/OpenRA.Game//Widgets/WidgetLoader.cs:68
  at OpenRA.WidgetLoader.LoadWidget (OpenRA.Widgets.WidgetArgs args, OpenRA.Widgets.Widget parent, OpenRA.MiniYamlNode node) [0x00152] in /Users/thill/projects/openra/OpenRA.Game//Widgets/WidgetLoader.cs:68
  at OpenRA.WidgetLoader.LoadWidget (OpenRA.Widgets.WidgetArgs args, OpenRA.Widgets.Widget parent, OpenRA.MiniYamlNode node) [0x00152] in /Users/thill/projects/openra/OpenRA.Game//Widgets/WidgetLoader.cs:68
  at OpenRA.WidgetLoader.LoadWidget (OpenRA.Widgets.WidgetArgs args, OpenRA.Widgets.Widget parent, System.String w) [0x00032] in /Users/thill/projects/openra/OpenRA.Game//Widgets/WidgetLoader.cs:43
  at OpenRA.Game.LoadWidget (OpenRA.World world, System.String id, OpenRA.Widgets.Widget parent, OpenRA.Widgets.WidgetArgs args) [0x00041] in /Users/thill/projects/openra/OpenRA.Game//Game.cs:120
  at OpenRA.Mods.Common.Widgets.Logic.LoadIngamePlayerOrObserverUILogic..ctor (OpenRA.Widgets.Widget widget, OpenRA.World world) [0x000a3] in /Users/thill/projects/openra/OpenRA.Mods.Common//Widgets/Logic/Ingame/LoadIngamePlayerOrObserverUILogic.cs:31
```

Load up A River Runs Near with AIs and I get this stacktrace.

File: OpenRA.Mods.Common/Widgets/RadarWidget.cs [hunk 1]
```diff
@@ -83,16 +93,39 @@ public override void Initialize(WidgetArgs args)
...
+			var right = map.Bounds.Right * cellWidth;
+
+			for (var x = projectedLeft; x < projectedRight; x++)
+			{
+				var allTop = map.Unproject(new PPos(x, projectedTop));
+				var allBottom = map.Unproject(new PPos(x, projectedBottom));
+				if (allTop.Any())
+					top = Math.Min(top, allTop.MinBy(uv => uv.V).V);
+
+				if (allTop.Any())
```
Comment (penev92) [inline OpenRA.Mods.Common/Widgets/RadarWidget.cs]: I bet this wants to be `allBottom` to avoid the crash @Phrohdoh posted.

**Comment by pchote** [comment]
Good catch.  Fixed.

**Comment by reaperrr** [comment]
Works :+1:

**Pull request merged**